tests/FlagConfigurationLoaderTest.php

```diff
@@ -24,112 +20,93 @@ class FlagConfigurationLoaderTest extends TestCase
-            ->with(self::FLAG_KEY)
-            ->willReturn($response['flags'][self::FLAG_KEY]);
+        $configStore->expects($this->once())
+            ->method('setFlags')->with($flags);
 
-        // Expect setConfig to not be called
-        $configStoreMock->expects($this->never())
-            ->method('setConfigurations');
+        $loader = new FlagConfigurationLoader($apiWrapper, $configStore);
+        $loader->fetchAndStoreConfigurations();
 
-        $fcl =  new FlagConfigurationLoader($httpClientMock, $configStoreMock);
 
-        $this->assertNotNull($fcl->getConfiguration(self::FLAG_KEY));
+        $flag = $loader->get(self::FLAG_KEY);
+        $this->assertInstanceOf(Flag::class, $flag);
+        $this->assertEquals(self::FLAG_KEY, $flag->key);
+        $this->assertEquals($flags[self::FLAG_KEY], $flag);
     }
 
-    /**
-     * @throws ClientExceptionInterface
-     * @throws InvalidApiKeyException
-     * @throws InvalidArgumentException
-     * @throws HttpRequestException
-     */
-    public function testThrows(): void
-    {
-        // Test that FlagLoader throws exceptions when appropriate
-        $ex =  new Exception('config requester error');
-        $response = json_decode(file_get_contents(self::MOCK_RESPONSE_FILENAME), true);
-        // Mock the webserver to return the hardcoded config above.
-        $flagLoader = $this->getFlagLoaderForData($response['flags'], $ex);
+    public function testLoadsOnGet() : void {
+        // Arrange: Load some flag data to be returned by the APIRequestWrapper
+        // Load mock response data
+        $flagsRaw = file_get_contents(self::MOCK_RESPONSE_FILENAME);
+        $flagsJson = json_decode($flagsRaw, true);
+        $flags = array_map(fn ($flag) => (new UFCParser())->parseFlag($flag), $flagsJson['flags']);
 
-        $this->expectException(Exception::class);
+        $apiWrapper = $this->getMockBuilder(APIRequestWrapper::class)->disableOriginalConstructor()->getMock();
 
-        $flag = $flagLoader->getConfiguration(self::FLAG_KEY);
+        $cache = DefaultCacheFactory::create();
+        // Act: Create a new FCL and retrieve a flag
+        $loader = new FlagConfigurationLoader($apiWrapper, new ConfigurationStore($cache));
+
+        // Mocks verify interaction of loader <--> API requests and loader <--> config store
+        $apiWrapper->expects($this->once())
+            ->method('get')
+            ->willReturn($flagsRaw);
+
+
+        $flag = $loader->get(self::FLAG_KEY);
+
+        // Assert: non-null flag, api called only once via Mock `expects` above.
+        $this->assertNotNull($flag);
     }
 
-    /**
-     * @param array $data
-     * @param Throwable|null $mockedThrowable
-     * @return FlagConfigurationLoader
-     */
-    private function getFlagLoaderForData(array $data, ?Throwable $mockedThrowable = null): FlagConfigurationLoader
-    {
-        $cache = new FileSystemCache();
+    public function testReloadsOnExpiredCache(): void {
```

Comment:
This test should start with some data in the cache, then reload, and make sure it's just the new data.

The new data should have one flag no longer present, one flag that carries over, and one new flag

Reply:
✔️